src/categories/categories.service.ts

```diff
@@ -65,4 +65,30 @@ export class CategoriesService {
       .exec();
     return doc;
   }
+
+  async pullMenuFromCategory(
+    categoryId: Types.ObjectId,
+    menuId: Types.ObjectId,
+  ) {
+    const doc = await this.categoryModel
+      .findByIdAndUpdate(
+        categoryId,
+        { $pull: { menus: menuId } },
+        { new: true },
+      )
+      .exec();
+    return doc;
+  }
+
+  async pullManyMenusFromCategories(
+    categoryIds: Types.ObjectId[],
+    menuIds: Types.ObjectId[],
+  ) {
+    await this.categoryModel
+      .updateMany(
+        { _id: { $in: categoryIds } },
```

Comment:
May I ask why this this update don't have `{new: true}` and why its not return like `pullMenuFromCategory`

Reply:
`{new: true}` is an option for queries with can return the document. Setting it to true means that it will return the updated document.

But sadly, `UpdateMany` doesn't return any document.

Reply:
So, this service is gonna be used by backend(us) not frontend And`pullMenuFromCategory` need to return the updated body to frontend, Am I right?.

Reply:
ok cool,TY